Add "d8 status" command for cluster diagnostics

VaLosev opened the pull request:
Added the d8 status command, which will help perform initial diagnostics of the cluster and will help in debugging closed environments.

```
> ./d8 status

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃      Cluster Status Report     ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
┌ Master Nodes Status
├ NAME                                           STATUS          VERSION        IP-ADRESS
└ losev-master-1                                 Ready           v1.30.10	10.0.99.78

┌ Deckhouse Releases
├ NAME          PHASE           TRANSITIONTIME   MESSAGE
├ v1.69.11	Superseded	21d
├ v1.69.13	Superseded	14d 22h
├ v1.69.14	Superseded	14d
├ v1.69.15	Superseded	12d 23h
└ v1.69.16	Deployed	12d 23h

┌ Deckhouse Pods
├ NAME                          READY    STATUS       RESTARTS   AGE
└ deckhouse-6d7595969b-mgxbn    2/2      Running      0          12d 23h

✅ No Cluster Alerts found

┌ Deckhouse Edition
└ EE

┌ Deckhouse Registry Information
├ Registry: registry.deckhouse.ru/deckhouse/ee
└ Scheme: https

┌ ModuleConfig Deckhouse Settings
├ releaseChannel: Stable
├ bundle: Default
└ logLevel: Info

┌ CNI in cluster
├ NAME               WEIGHT   SOURCE     PHASE        ENABLED  READY
├ cni-cilium         21       Embedded   Ready        True     True
├ cni-flannel        35       Embedded   Downloaded   False    False
└ cni-simple-bridge  35       Embedded   Downloaded   False    False

┌ Deckhouse Queue
Summary:
- 'main' queue: empty.
- 114 other queues (0 active, 114 empty): 0 tasks.
- no tasks to handle.
```

VaLosev committed:
add d8 status V1

Signed-off-by: Losev Valery <valery.losev@flant.com>

VaLosev committed:
add d8 status V2

Signed-off-by: Losev Valery <valery.losev@flant.com>

VaLosev committed:
add d8 status V2.1

Signed-off-by: Losev Valery <valery.losev@flant.com>

VaLosev committed:
added comments + fixed bugs

Signed-off-by: Losev Valery <valery.losev@flant.com>

VaLosev commented:
Tested on several clusters, corrected the errors I found.
Added comments.
```
d8 status --help
Get status information about Deckhouse Kubernetes Platform

 © Flant JSC 2025

Usage:
  d8 status [flags]

Flags:
      --context string      The name of the kubeconfig context to use
  -h, --help                help for status
  -k, --kubeconfig string   KubeConfig of the cluster. (default is $KUBECONFIG when it is set, $HOME/.kube/config otherwise) (default "...")
```

mvasl commented on internal/status/cmd/status.go:
```diff
@@ -0,0 +1,102 @@
+
+© Flant JSC 2025`)
+
+func NewCommand() *cobra.Command {
+    statusCmd := &cobra.Command{
+	Use:           "status",
+	Short:         "Get cluster status information",
+	Long:          statusLong,
+	SilenceErrors: true,
+	SilenceUsage:  true,
+	RunE:          runStatus,
+    }
```
`go fmt`, please

mvasl commented on internal/status/commands/cni_modules/cnimodules.go:
```diff
@@ -0,0 +1,137 @@
+    var sb strings.Builder
+    yellow := color.New(color.FgYellow).SprintFunc()
+    sb.WriteString(yellow("┌ CNI in cluster:\n"))
+    sb.WriteString(yellow(fmt.Sprintf("├ %-18s %-8s %-10s %-12s %-8s %-8s\n", "NAME", "WEIGHT", "SOURCE", "PHASE", "ENABLED", "READY")))
+    for i, module := range modules {
+        prefix := "├"
+        if i == len(modules)-1 {
+            prefix = "└"
+        }
+        name := module.Name
+        weight := module.Weight
+        source := module.Source
+        phase := module.Phase
+        enabled := module.Enabled
+        ready := module.Ready
+        sb.WriteString(fmt.Sprintf("%s %-18s %-8s %-10s %-12s %-8s %-8s\n",
+            yellow(prefix),
+            name,
+            weight,
+            source,
+            phase,
+            enabled,
+            ready,
+        ))
+    }
+    return sb.String()
```
You can use [text/tabwriter](https://pkg.go.dev/text/tabwriter) to avoid hardcoding lengths of columns everywhere

mvasl commented on internal/status/commands/releases/releases.go:
```diff
@@ -0,0 +1,126 @@
+        }
+        timeAgo := ""
+        transitionTimeValue := release.TransitionTime
+        if transitionTimeValue != "" {
+            if transitionTime, err := time.Parse(time.RFC3339, transitionTimeValue); err == nil {
+                duration := time.Since(transitionTime)
+                days := int(duration.Hours()) / 24
+                hours := int(duration.Hours()) % 24
+                if days > 0 {
+                    timeAgo = fmt.Sprintf("%dd", days)
+                }
+                if hours > 0 {
+                    if timeAgo != "" {
+                        timeAgo += " "
+                    }
+                    timeAgo += fmt.Sprintf("%dh", hours)
+                }
+            } else {
+                timeAgo = "Parse Error"
+            }
+        }
```
We talked about "[Return early pattern](https://habr.com/ru/articles/348074/)", this is a good example of it's value. If your code goes into the branch on line 100, it will not print anything to `timeAgo` but it is hard to spot that behind all of this surrounding logic.
<img width="545" height="554" alt="image" src="https://github.com/user-attachments/assets/7ce11cb6-5f46-4341-a807-d423406f18f9" />

VaLosev committed:
add d8 status V3

Signed-off-by: Losev Valery <valery.losev@flant.com>

name212 commented on internal/status/commands/pods/deckhousepods.go:
```diff
@@ -0,0 +1,135 @@
+	}
+}
+
+func formatAge(t time.Time) string {
```
we use same logic in many places. I think we can move it to separate package and use one function in all places

name212 commented on internal/status/cmd/status.go:
Please add license to all files

name212 reviewed (changes requested):
some nits

VaLosev committed:
add d8 status V4

Signed-off-by: Losev Valery <valery.losev@flant.com>

VaLosev commented:
Made changes:
- All additional packages have been moved to the tools section. ([tools](https://github.com/deckhouse/deckhouse-cli/tree/feature/add-status-commands/internal/status/tools))
- Made the queue receiving function execution in a separate package. (`...internal/status/tools/queuefetcher`)
- Moved the time format processing functions to a separate package. (`...internal/status/tools/timeutil`)
- The main functions are now in `.../status/objects`
- Added a license comment to all files.
- Only used variables are left in the constants file.
- Added a check for the `SEVERITY` field in `ClusterAlert` so that an empty field does not cause panic. Now, if a custom alert does not contain SEVERITY, N/A will be substituted.
